psi4/driver/p4util/python_helpers.py

```diff
@@ -75,7 +75,16 @@ def pybuild_basis(mol, key=None, target=None, fitrole='ORBITAL', other=None, pur
         core.print_out(basisdict['message'])
 
     psibasis = core.BasisSet.construct_from_pydict(mol, basisdict, puream)
-    return psibasis
+    ecpbasis = None
+    if 'ecp_shell_map' in basisdict:
+        ecpbasis = core.BasisSet.construct_ecp_from_pydict(mol, basisdict, puream)
+
+    if key == 'BASIS':
+        # For orbitals basis sets, we need to return ECP also
+        return psibasis, ecpbasis
```

Comment:
Much of this stuff is the result of me spending a few days trying to understand the python basis set stuff, and failing miserably.  The full test suite passes as-is, so I think I caught the majority of any damage done by my py-side changes.  I don't disagree about stuffing the ECP stuff inside basis set objects if needed, but I'm reluctant to do that without feedback from @loriab, who wrote the py-side infrastructure.  My intent with this PR is to get something in that works, so that we can focus on a proper derivative implementation, with a basis set machinery cleanup pass to follow.  I'm willing to reconsider that approach if you have a better idea.